### ISSUE TITLE
linter: skip jsx-a11y checks on element-based composition props (`asChild` / `render`) via config

### ISSUE THREAD

**Issue by tobigumo** [oxc-project/oxc#23645]
### Summary
With jsx-a11y, element-based composition (Radix `asChild`, Base UI / shadcn `render`) produces false positives such as `anchor-has-content` and `control-has-associated-label` on valid code. I've opened a design issue upstream: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/1091. This isn't about rule-behavior parity; it's about which nodes are evaluated (engine / config level), so oxc could implement it ahead of upstream.

### Minimal reproduction
```tsx
<Button render={<a href="/x" />}>Label</Button>
// false positive: anchor-has-content on <a/>
// (at runtime the wrapping component's children are forwarded → <a>Label</a>)

<Button asChild aria-label="Delete"><a href="/x" /></Button>
// false positive (at runtime the aria-label is forwarded to the child)
```
The wrapping component and the composed element merge into one element at runtime, but the linter evaluates them as two separate nodes.

### Request
Add an optional setting to declare composition props, so a component using one of them, together with its composed element, is excluded from a11y evaluation:
```jsonc
"settings": { "jsx-a11y": { "compositionProps": ["asChild", "render"] } }
```
It follows the same pattern as `polymorphicPropName` / `components`. It is the user's responsibility to declare only components that actually forward props.

Fully resolving the merged element could work in theory, but prop-merge semantics are library-specific (e.g. which value wins on conflict), so they can't be resolved generically. This issue only asks for the skip behavior.

### Why oxc could lead here
- This concerns which nodes are evaluated and how related nodes are grouped (the engine/config layer), not rule judgment, so it can be added as an oxc-specific setting without breaking upstream parity.
- `eslint-plugin-jsx-a11y` has a slower release cadence, while element-based composition (Radix / Base UI / shadcn) is already widespread, so users are hitting this today.

### Optional follow-up (not required)
For rules that check for the presence of content or a label (`anchor-has-content`, `control-has-associated-label`, `alt-text`), these false positives could be *resolved* rather than skipped by checking the union of the wrapping component and the composed element. Forwarding is the contract of these patterns, and presence doesn't depend on which value wins a merge, so this stays valid without modeling any library-specific merge rules. Skipping alone already removes the false positives, so this is just a possible future enhancement.
